File: client/projects/data/src/lib/models/carousel-item.ts
import { CarouselType } from '../enums';
import { CarouselItemPhoto } from './carousel-item-photo';

export interface CarouselItem {
  id: number;
  type: CarouselType;
  sort: number;
  title: string;
  subTitle: string;
  body: string;
  isVisible: boolean;
  primaryButtonLabel: string;
  primaryButtonUrl: string;
  secondaryButtonLabel: string;
  secondaryButtonUrl: string;
  photo: CarouselItemPhoto;
}

File: server/Audi/Entities/CarouselItem.cs
using System;
using System.Collections.Generic;

namespace Audi.Entities
{
    public class CarouselItem
    {
        public int Id { get; set; }
        public string Type { get; set; }
        public int Sort { get; set; }
        public string Title { get; set; }
        public string SubTitle { get; set; }
        public string Body { get; set; }
        public bool IsVisible { get; set; } = false;
        public string PrimaryButtonLabel { get; set; }
        public string PrimaryButtonUrl { get; set; }
        public string SecondaryButtonLabel { get; set; }
        public string SecondaryButtonUrl { get; set; }
        public int PhotoId { get; set; }
        public CarouselItemPhoto Photo { get; set; }
        public ICollection<HomepageCarouselItem> HomepageCarouselItems { get; set; }
    }
}